src/Support/InferExtensions/ResponderTypeInfer.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Dedoc\Scramble\Support\InferExtensions;
+
+use Dedoc\Scramble\Infer\Definition\ClassDefinition;
+use Dedoc\Scramble\Infer\Extensions\ExpressionTypeInferExtension;
+use Dedoc\Scramble\Infer\Scope\Scope;
+use Dedoc\Scramble\Support\Type\ObjectType;
+use Dedoc\Scramble\Support\Type\Type;
+use League\Fractal\TransformerAbstract;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use Dedoc\Scramble\Support\Generator\Types\UnknownType;
+use Dedoc\Scramble\Infer\Services\FileNameResolver;
+use Dedoc\Scramble\Support\ResponseExtractor\ModelInfo;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Str;
+
+class ResponderTypeInfer implements ExpressionTypeInferExtension
+{
+    private $modelType;
+    private string $modelPropertyName;
+    public static $transformerModelTypesCache = [];
+
+    public function getType(Expr $node, Scope $scope): ?Type
+    {
+        if (
+            !$scope->classDefinition()?->isInstanceOf(TransformerAbstract::class)
+            || $scope->classDefinition()?->name === TransformerAbstract::class
+        ) {
+            return null;
+        }
+
+        $transformerArguments = $scope->classDefinition()->getMethodDefinition('transform')->type->arguments;
```

Comment:
Just to have a more readable code :)
```suggestion
        $transformerArguments = $scope
            ->classDefinition()
            ->getMethodDefinition('transform')
            ->type
            ->arguments;
```